client/main.rs

```diff
@@ -60,8 +70,18 @@ struct Params {
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     env_logger::Builder::from_env(Env::default().default_filter_or("info")).init();
+
     let params = Params::from_args();
 
+    let (grpc_urls_channel_sender, mut grpc_urls_channel_receiver) =
+        tokio::sync::mpsc::channel::<Vec<String>>(1);
+
+    let mut grpc_urls_from_file = read_grpc_urls_from_file(params.grpc_urls_file).unwrap();
```

Comment:
better return error than making `unwrap`

Reply:
If this fine?
```
    let mut grpc_urls_from_file;

    if let Ok(urls) = read_grpc_urls_from_file(params.grpc_urls_file) {
        grpc_urls_from_file = urls;
    } else {
        panic!("Failed to read grpc urls from file")
    }
```